### ISSUE TITLE
[Feature] argmax / argmin reduction operations

### ISSUE THREAD

**Issue by brody-0125** [brody-0125/dart_tensor_preprocessing#34]
## Summary

Add `argmax` and `argmin` operations to find indices of maximum/minimum values along specified axes.

## Motivation

`argmax` and `argmin` are essential for decoding model outputs (e.g., getting the predicted class index from a softmax output). This is one of the most commonly needed post-processing operations after model inference.

## Reference Implementations

- **PyTorch**: `torch.argmax(input, dim=None)` — [PyTorch docs](https://pytorch.org/docs/stable/generated/torch.argmax.html)
- **NumPy**: `numpy.argmax(a, axis=None)` — [NumPy docs](https://numpy.org/doc/stable/reference/generated/numpy.argmax.html)
- **ONNX**: `ArgMax` operator (opset 1+) — [ONNX spec](https://onnx.ai/onnx/operators/onnx__ArgMax.html)

## Verification

- ✅ `torch.argmax` exists — VERIFIED
- ✅ ONNX `ArgMax` operator exists in opset 1+
- ✅ Feature is MISSING from dart_tensor_preprocessing — VERIFIED (searched for argmax, argmin, argMax, argMin)
- Note: `TensorBufferReduce` has `min()`, `max()`, `minAxis()`, `maxAxis()` but NOT their arg variants

## Proposed API

```dart
// Global argmax (returns flat index)
final idx = tensor.argmax();

// Axis-wise argmax (returns tensor of indices)
final indices = tensor.argmaxAxis(axis: 1);

// As TransformOp for pipeline integration
final op = ArgMaxOp(axis: 1, keepDims: false);
final result = op.apply(tensor);
```

## Implementation Notes

- Extend `TensorBufferReduce` extension with `argmax()`, `argmin()`, `argmaxAxis()`, `argminAxis()`
- Also create `ArgMaxOp` / `ArgMinOp` as `TransformOp` subclasses for pipeline usage
- Support `keepDims` parameter (like PyTorch/ONNX)
- Output dtype should be `DType.int64` (matching ONNX ArgMax output type)
- Handle strided (non-contiguous) tensors
- Add `OperationCapabilities`:
  - `onnxOpType: 'ArgMax'` / `'ArgMin'`
  - `onnxOpsetVersion: 1`
  - `pytorchEquivalent: 'torch.argmax'`

## ROI Assessment

**Very High** — Without argmax, users cannot decode classification model outputs. This is the most basic post-processing operation needed after running any classification model through ONNX Runtime.
